numpy/matlib.py

```diff
@@ -1,6 +1,19 @@
+import warnings
+
+# 2018-05-29, PendingDeprecationWarning added to matrix.__new__
+# 2020-01-23, numpy 1.19.0 DeprecatonWarning
+warnings.warn("Importing from numpy.matlib is deprecated since 1.19.0. "
+              "The matrix subclass is not the recommended way to represent "
+              "matrices or deal with linear algebra (see "
+              "https://docs.scipy.org/doc/numpy/user/numpy-for-matlab-users.html). "
+              "Please adjust your code to use regular ndarray. ",
+              DeprecationWarning, stacklevel=2)
```

Comment:
Looks to me like this should be `PendingDeprecationWarning`, same as `np.matrix` itself. This module contains `matrix` construction functions that can't be found elsewhere. There's no reason to deviate in deprecation strategy here.

Reply:
@mattip said "add a DeprecationWarning" so I added a DeprecationWarning,

It's easy to change either way.

https://github.com/numpy/numpy/pull/10142 starts with "As discussed on the mailing list." so I think this deprecation has been discussed on the list.